tests/Provider/BaseProviderThisVerOrBelowTest.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Tests\Provider;
+
+use Gaufrette\Adapter;
+use Gaufrette\File;
+use Gaufrette\Filesystem;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\MediaBundle\CDN\CDNInterface;
+use Sonata\MediaBundle\CDN\Server;
+use Sonata\MediaBundle\Generator\DefaultGenerator;
+use Sonata\MediaBundle\Metadata\MetadataBuilderInterface;
+use Sonata\MediaBundle\Model\MediaInterface;
+use Sonata\MediaBundle\Provider\BaseProvider;
+use Sonata\MediaBundle\Tests\Entity\Media;
+use Sonata\MediaBundle\Thumbnail\ThumbnailInterface;
+use Symfony\Component\Form\FormBuilder;
+
+class BaseProviderThisVerOrBelowTest extends AbstractProviderTest
+{
+    public function getProvider()
+    {
+        $adapter = $this->createMock(Adapter::class);
+
+        $filesystem = $this->getMockBuilder(Filesystem::class)
+            ->setMethods(['get'])
+            ->setConstructorArgs([$adapter])
+            ->getMock();
+        $file = $this->getMockBuilder(File::class)
+            ->setConstructorArgs(['foo', $filesystem])
+            ->getMock();
+
+        $cdn = new Server('/uploads/media');
+
+        $generator = new DefaultGenerator();
+
+        $thumbnail = $this->createMock(ThumbnailInterface::class);
+
+        $metadata = $this->createMock(MetadataBuilderInterface::class);
+
+        $provider = new TestBaseProvider('test', $filesystem, $cdn, $generator, $thumbnail, $metadata);
+        $this->assertInstanceOf(BaseProvider::class, $provider);
+
+        return $provider;
+    }
+
+    public function testPostRemove(): void
+    {
+        $provider = $this->getProvider();
+        $media = new Media();
+        
+        //The ID is more than a thousand, the bug manifests itself under this condition.
```

Comment:
You can remove this comment, as the context of *the bug* is missing after merging this PR.

Reply:
This test is more likely to demonstrate the error before correction and its absence after. In the resulting branch, it is preferable to delete it completely.